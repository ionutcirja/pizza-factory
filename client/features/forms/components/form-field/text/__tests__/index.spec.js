import React from 'react';
import { shallow } from 'enzyme';
import TextField from '..';

describe('TextField component', () => {
  let propsToRender;
  
  beforeEach(() => {
    propsToRender = {
      field: {
        name: 'id',
        value: '',
      },
      form: {
        touched: {},
        errors: {},
      },
      placeholder: 'search',
    };
  });
  
  describe('render', () => {
    it('should render a text input', () => {
      const wrapper = shallow(<TextField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render a disabled text input if disabled prop value is truthy', () => {
      propsToRender.disabled = true;
      const wrapper = shallow(<TextField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render an error if field is touched and'
      + ' there is an error defined for the current field', () => {
      propsToRender.form = {
        touched: {
          id: true,
        },
        errors: {
          id: 'some error',
        },
      };
      const wrapper = shallow(<TextField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
