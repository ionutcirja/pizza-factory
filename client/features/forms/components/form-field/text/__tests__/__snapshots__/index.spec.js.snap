// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component render should render a disabled text input if disabled prop value is truthy 1`] = `
<FieldSet>
  <Label
    colour="grey"
    htmlFor="id"
  >
    label
  </Label>
  <Input
    borderColour="grey"
    borderErrorColour="red"
    colour="black"
    disabled={true}
    id="id"
    name="id"
    placeholder="search"
    type="text"
    value=""
  />
</FieldSet>
`;

exports[`TextField component render should render a text input 1`] = `
<FieldSet>
  <Label
    colour="grey"
    htmlFor="id"
  >
    label
  </Label>
  <Input
    borderColour="grey"
    borderErrorColour="red"
    colour="black"
    id="id"
    name="id"
    placeholder="search"
    type="text"
    value=""
  />
</FieldSet>
`;

exports[`TextField component render should render an error if field is touched and there is an error defined for the current field 1`] = `
<FieldSet>
  <Label
    colour="grey"
    htmlFor="id"
  >
    label
  </Label>
  <Input
    borderColour="grey"
    borderErrorColour="red"
    colour="black"
    error="some error"
    id="id"
    name="id"
    placeholder="search"
    type="text"
    value=""
  />
  <Error
    colour="red"
  >
    some error
  </Error>
</FieldSet>
`;
